### ISSUE TITLE
Recurring jobs next execution issue after updating cron expression in >=1.8.3

### ISSUE THREAD

**Issue by taskbit** [HangfireIO/Hangfire#2256]
Our application updates existing recurring jobs to use `Cron.Never()` on startup, to temporarily prevent scheduling.
Recurring jobs later are being updated e.g. to use `Cron.Minutely()`

We noticed in Hangfire versions >=1.8.3, recurring jobs next execution state is always "DISABLED" after updating from Cron.Never to something specific. While the cron expression update is reflected correctly, the job is never scheduled (see screenshots below).

This was working as expected in 1.8.2 but now leads to the recurring job never being scheduled:

```
IRecurringJobManager.AddOrUpdate(id, methodCall, Cron.Never());
IRecurringJobManager.AddOrUpdate(id, methodCall, Cron.Minutely());
```

Minimal working example to reproduce the issue (.NET 7, Hangfire 1.8.4)

```
using System.Linq.Expressions;
using Hangfire;

var builder = WebApplication.CreateBuilder(args);

builder.Services.AddControllers();
builder.Services.AddHangfire(c =>
{
    c.UseSqlServerStorage("CONNECTION_STRING");
});
builder.Services.AddHangfireServer();
builder.Services.AddTransient<JobManager>();

var app = builder.Build();

app.UseHangfireDashboard();
app.MapControllers();

var jobManager = app.Services.GetRequiredService<JobManager>();
jobManager.ScheduleRecurring<RecurringJob>("RecurringJob", j => j.Execute(), Cron.Never());
jobManager.ScheduleRecurring<RecurringJob>("RecurringJob", j => j.Execute(), Cron.Minutely());

app.Run();

class JobManager
{
    private readonly IRecurringJobManager _recurringJobManager;

    public JobManager(IRecurringJobManagerFactory recurringJobManagerFactory)
    {
        _recurringJobManager = recurringJobManagerFactory.GetManager(JobStorage.Current);
    }

    public void ScheduleRecurring<T>(string id, Expression<Func<T, Task>> methodCall, string cronExpression)
    {
        _recurringJobManager.AddOrUpdate(id, methodCall, cronExpression);
    }
}

class RecurringJob
{
    public Task Execute()
    {
        return Task.CompletedTask;
    }
}
```

<img width="916" alt="hangfire_182" src="https://github.com/HangfireIO/Hangfire/assets/695343/e175e1bb-726d-4451-b3ff-e38fd62efca7">

<img width="911" alt="hangfire_184" src="https://github.com/HangfireIO/Hangfire/assets/695343/e52155c4-6c79-46e5-9637-13bc1d76be14">

**Comment by cezar-pimentel** [comment]
Yes, I can confirm that. I noticed the same behavior on my last implemented project.

Will try to look further into the code and try to figure out how to address this issue.

**Comment by odinserj** [comment]
Thanks for reporting this regression, @taskbit! It will be fixed in Hangfire 1.8.5 soon.
